### PR TITLE
optimize: optimize recognize sql in limit and order by 

### PR DESCRIPTION
**Pull request by l81893521**
### Ⅰ. Describe what this PR did
This solution still has a problem that 
```
getLimitCondition()
```
and
```
getOrderByCondition()
```
in the 
```
WhereRecognizer.java 
```
and I will refactor it in the later version.

### CODE REVIEW

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/postgresql/BasePostgresqlRecognizer.java [hunk 1]
```diff
@@ -82,4 +85,26 @@ public String getWhereCondition(SQLExpr where) {
...
+        return sb.toString();
+    }
+
+    protected String getOrderByCondition(SQLOrderBy sqlOrderBy) {
+        if (Objects.isNull(sqlOrderBy)) {
+            return StringUtils.EMPTY;
+        }
+
+        StringBuilder sb = new StringBuilder();
+        executeOrderBy(sqlOrderBy, new MySqlOutputVisitor(sb));
```
Comment (caohdgege) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/postgresql/BasePostgresqlRecognizer.java:106]: 不是 PGOutputVisitor 吗？

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/mysql/MySQLDeleteRecognizer.java [hunk 2]
```diff
@@ -94,13 +96,15 @@ public String getWhereCondition() {
     }
 
     @Override
-    public String getLimit() {
-        return super.getLimit(ast, getSQLType());
+    public String getLimitCondition() {
+        SQLLimit limit = ast.getLimit();
+        return super.getLimitCondition(limit);
```
Comment (caohdgege) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/mysql/MySQLDeleteRecognizer.java:101]: 这里需要判断是否包含占位符，如果有的话，需要把对应的值加入到paramAppenderList里面

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/UpdateExecutor.java [hunk 3]
```diff
@@ -81,13 +81,13 @@ private String buildBeforeImageSQL(TableMeta tableMeta, ArrayList<List<Object>>
         if (StringUtils.isNotBlank(whereCondition)) {
             suffix.append(WHERE).append(whereCondition);
         }
-        String orderBy = recognizer.getOrderBy();
+        String orderBy = recognizer.getOrderByCondition();
         if (StringUtils.isNotBlank(orderBy)) {
-            suffix.append(orderBy);
+            suffix.append(" ").append(orderBy);
```
Comment (a364176773) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/UpdateExecutor.java:86]: “ ” you can set it to a constant, right?

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/mysql/MySQLDeleteRecognizer.java (hunk 2 above)
Comment (a364176773) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/mysql/MySQLDeleteRecognizer.java:101]: +1

Comment (l81893521) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/mysql/MySQLDeleteRecognizer.java:101]: fixed

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/postgresql/BasePostgresqlRecognizer.java (hunk 1 above)
Comment (l81893521) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/postgresql/BasePostgresqlRecognizer.java]: fixed

**Comment by caohdgege** [comment]
plz modify AntlrMySQLDeleteRecognizer, AntlrMySQLSelectRecognizer, AntlrMySQLUpdateRecognizer either

**Comment by l81893521** [comment]
> plz modify AntlrMySQLDeleteRecognizer, AntlrMySQLSelectRecognizer, AntlrMySQLUpdateRecognizer either

fixed

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by caohdgege — APPROVED** [review]
LGTM

**Review by slievrly — COMMENTED** [review]
LGTM

**Pull request merged**